app/src/main/java/com/android/streetworkapp/MainActivity.kt

```diff
@@ -9,26 +9,75 @@ import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
 import androidx.navigation.compose.rememberNavController
 import androidx.navigation.navigation
+import com.android.streetworkapp.model.event.Event
+import com.android.streetworkapp.model.event.EventList
+import com.android.streetworkapp.model.park.Park
+import com.android.streetworkapp.model.parklocation.OverpassParkLocationRepository
+import com.android.streetworkapp.model.parklocation.ParkLocationViewModel
+import com.android.streetworkapp.ui.map.MapScreen
 import com.android.streetworkapp.ui.navigation.NavigationActions
 import com.android.streetworkapp.ui.navigation.Route
 import com.android.streetworkapp.ui.navigation.Screen
+import com.android.streetworkapp.ui.park.ParkOverview
 import com.android.streetworkapp.ui.profile.ProfileScreen
+import com.google.firebase.Timestamp
+import okhttp3.OkHttpClient
 
 class MainActivity : ComponentActivity() {
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     Log.d("MainActivity", "Setup content")
-    setContent(parent = null) {}
+    setContent(parent = null) { StreetWorkAppMain() }
   }
 }
 
 // the testInvokation is super ugly but I have NOT found any other way to test the navigation from a
 // ui perspective since we don't use fragments
 @Composable
 fun StreetWorkAppMain(testInvokation: NavigationActions.() -> Unit = {}) {
+
+  // repositories
+  val overpassParkLocationRepo = OverpassParkLocationRepository(OkHttpClient())
+  // viewmodels
+  val parkLocationViewModel = ParkLocationViewModel(overpassParkLocationRepo)
+
+  StreetWorkApp(parkLocationViewModel, testInvokation)
+}
+
+@Composable
+fun StreetWorkApp(
+    parkLocationViewModel: ParkLocationViewModel,
+    navTestInvokation: NavigationActions.() -> Unit = {},
+    mapCallbackOnMapLoaded: () -> Unit = {}
+) {
   val navController = rememberNavController()
   val navigationActions = NavigationActions(navController)
 
+  val eventList =
+      EventList(
+          events =
+              listOf(
+                  Event(
+                      eid = "1",
+                      title = "Group workout",
+                      description = "A fun group workout session to train new skills",
+                      participants = 3,
+                      maxParticipants = 5,
+                      date = Timestamp(0, 0), // 01/01/1970 00:00
+                      owner = "user123")))
+
+  // Park with events
+  val testPark =
+      Park(
+          pid = "1",
+          name = "EPFL Esplanade",
+          location = "EPFL",
+          image = null,
+          rating = 4.5f,
+          nbrRating = 102,
+          occupancy = 0.8f,
+          events = eventList)
+
   NavHost(
       navController = navController,
       startDestination = Route.AUTH) { // TODO: handle start destination based on signIn logic
```

Comment:
Proposition: In order to have a runnable demo for the next coaching meeting, I think it could be useful to have a temporary starting screen, to make sure that the main is directly runnable. 
Here is a simple exemple:

```
composable(Screen.AUTH) {
            Column(
                modifier = Modifier.fillMaxSize(),
                verticalArrangement = Arrangement.Center,
                horizontalAlignment = Alignment.CenterHorizontally) {
                  Button(onClick = { navigationActions.navigateTo(Screen.MAP) }) { Text("Start") }
                }
          }
```

Reply:
Good idea, I implemented this but changed `navigationActions.navigateTo(Screen.MAP)` to `navigationActions.navigateTo(Route.MAP)` to clear the backstack